### DISCUSSION TITLE
Scaled shapes

### DISCUSSION THREAD

**Discussion by LeXXik** [jrouwe/JoltPhysics.js#68] [Q&A] [answered]
I am looking into the scaled shapes. They seem to work well for something like boxes - instead of defining half extents, I can give it a unit half extents and then control the shape via the scale. Pretty convenient. From what I read, the non-uniform and negative scales are supported. Are there caveats in using a scaled shape? For example, a non-uniformly scaled sphere is acting up:

https://github.com/jrouwe/JoltPhysics.js/assets/5677782/9e37fa01-62ef-45d7-b944-3cbef2fbd5f7

**Comment by jrouwe** [comment, marked as answer]
I just added some extra documentation here:

https://jrouwe.github.io/JoltPhysics/class_shape.html#a705685386feede5b6e0473ee00f5cdc1

And I also exposed the Shape::IsScaleValid / ScaleShape functions in Jolt.idl (didn't release a new version yet, not sure if you directly need these functions as you can easily avoid errors).

**Reply by LeXXik** [reply]
Thank you! This clarifies it. Going to add some guards in the debug version.
